File: daily/1047. 删除字符串中的所有相邻重复项.js
/**
 * @param {string} S
 * @return {string}
 */
var removeDuplicates = function(S) {
    var stack=[];
    for(i of S){
        let top=stack.pop();
        if(i!=top){ //不相等就按之前顺序压回栈中
            stack.push(top);
            stack.push(i);
        }
    }
    return stack.join('');
};

File: daily/C++版本/1047. 删除字符串中的所有相邻重复项.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        stack<char> stack;
        string res="";
        for(int i=0;i<S.length();i++){
            if(!stack.empty() && S[i]==stack.top()){
                    stack.pop();  
                    res.pop_back();              
            }
            else{
                stack.push(S[i]);
                res+=S[i];
            }
        }
        return res;
    }
};